File: backends/spring-boot/src/main/java/com/hellotime/controller/HealthController.java
package com.hellotime.controller;

import com.hellotime.dto.ApiResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.Instant;
import java.util.Map;

@RestController
@RequestMapping("/api/v1")
public class HealthController {

    @GetMapping("/health")
    public ApiResponse<Map<String, Object>> health() {
        return ApiResponse.ok(Map.of(
                "status", "UP",
                "timestamp", Instant.now().toString(),
                "techStack", Map.of(
                        "framework", "Spring Boot 3.2.5",
                        "language", "Java 21",
                        "database", "SQLite"
                )
        ));
    }
}

File: backends/nest/src/health/health.controller.ts
/**
 * 健康检查控制器 — 所有 9 个后端共享同一健康检查契约
 *
 * 返回 { status, timestamp, techStack: { framework, language, database } }
 * 前端和验证脚本通过此端点确认后端存活和技术栈标识。
 */
import { Controller, Get } from '@nestjs/common';
import { ok } from '../common/dto/api-response.dto';
import { HealthService } from './health.service';

@Controller('health')
export class HealthController {
  constructor(private readonly healthService: HealthService) {}

  @Get()
  getHealth() {
    return ok(this.healthService.getHealth());
  }
}
